samples/generic-synthetic-broken-links/brokenLinkChecker.js

```diff
@@ -0,0 +1,136 @@
+const puppeteer = require('puppeteer');
+
+/**
+ * Creates a dictionary of links that have a non-2xx status code (mapped to their respective status codes).
+ * @param {Array} An array of link objects with targetUrl and statusCode properties.
+ * @throws {Error} If any links have a non-2xx status code.
+ */
+function createNon2xxLinksDictionary(linksResult) {
```

Comment:
I think that we may want to flip our mindset on this function a bit. this function doesnt create anything, it returns void, or throws an error in the case that a non-200 link is found.

I would consider changing the name of this to something like checkForNon2xxLinks(args)